test/functional/toolchains/qbs/test_qbs.py

```diff
@@ -100,3 +101,46 @@ def build(self):
 
     client.run("create .")
     assert "--products hello,hello" in client.out
+
+
+@pytest.mark.tool("qbs")
+def test_qbs_multiple_configurations():
+    client = TestClient()
+
+    context = {
+        "name": "hello",
+        "version": "2.0",
+        "package_name": "hello"
+    }
+
+    conanfile = textwrap.dedent('''
+    import os
+
+    from conan import ConanFile
+    from conan.tools.qbs import Qbs
+
+    class Recipe(ConanFile):
+        name = "hello"
+        version = "1.0"
+
+        exports_sources = "*.cpp", "*.h", "*.qbs"
+        settings = "os", "compiler", "arch", "build_type"
+
+        def build(self):
+            qbs = Qbs(self)
+            qbs.add_configuration("release", {"qbs.debugInformation": False})
+            qbs.add_configuration("debug", {"qbs.debugInformation": True})
+            qbs.build()
```

Comment:
This might be one of those cases that you want to remove the ``settings = "build_type"``, because this will be mostly ignored and build those two. 
But depending on how ``QbsDeps`` manages multi-configuration packages, this might be a challenge and will not work in some platforms (as Windows), because every ``conan install/create`` basically instantiates 1 configuration (either Release or Debug, but not both). So it is not straightforward to have both Debug and Release dependencies unless all the dependencies did package Release and Debug in their packages and removed ``build_type`` from settings. But this doesn't happen for any package in ConanCenter, so users wanting to use third party libraries from ConanCenter won't be able to use this in most cases.

Reply:
Ok, I got it for release/debug builds, but how does Conan handles multiple configuration builds e.g. when a binary built with different flags, like defines?
Will there be an option for that and essentially there will be 2 packages with different options?
Or can we built a package with 2 binaries with different flags? I can think of a full/light versions of a library.
It would help to have a real use-case for this feature rather than synthetic test.

Reply:
There are many dimensions and features in which Conan can model the package binaries and the binary compatibility. 

For example, changing flags with ``tools.build:cxxflags`` can alter the binaries or not. Uses can use the ``tools.info.package_id:confs`` to decide which ``confs`` are made part of the ``package_id``. Changing some ``-Werror`` will not change the binaries, but some other might change. 

But it might be better to try to model the binaries at a higher level of abstraction:
- conan has custom ``settings``, that can be extended to model different "variants" the user want to model. Not only to create new settings or subsettings, but also to extend existing ones with new values.
- conan has ``options``, recipes can decide to add custom variants for themselves

Exactly because the number of different binaries, the approach of packaging different binaries inside the same ``package_id`` doesn't scale nicely, and Conan recommends for most cases 1 binary variant inside each ``package_id`` value.